ArticleTemplates/assets/js/modules/messenger.js

```diff
@@ -0,0 +1,198 @@
+/* global Promise */
```

Comment:
The strategy to data has been to hold back on using 'newer' features of JS. However I'd be keen to move away from that and look at options that allow us to use these features, for example we could look at polyfilling, it may be that we do this on-demand for now and polyfill Promises. I'd be happy with this if we want to ensure full-support.